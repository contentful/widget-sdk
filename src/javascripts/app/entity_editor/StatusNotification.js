import React from 'react';
import PropTypes from 'prop-types';
import { sortBy } from 'lodash';
import { joinWithAnd } from 'utils/StringUtils';
import { Note, TextLink } from '@contentful/forma-36-react-components';
import { DocumentStatus } from '@contentful/editorial-primitives';

/**
 * This component renders an alert depending on the document status.
 *
 * The available status IDs are defined in `data/document/statusCode`.
 */

const formatLocales = ([first, second, third, ...rest]) => {
  let locales;
  if (rest.length) {
    const others = rest.length > 1 ? 'others' : 'other';
    locales = [first, second, third, `${rest.length} ${others}`];
  } else {
    locales = [first, second, third].filter(Boolean);
  }
  return joinWithAnd(locales);
};

const sortLocales = (locales) => sortBy(locales, [(l) => !l.default, 'internal_code']);

const formatErroredLocales = (erroredLocales) => {
  const locales = sortLocales(erroredLocales).map((l) => l.name);
  return formatLocales(locales);
};

const StatusCodeNotification = ({ status, entityLabel, erroredLocales, entityHref }) => {
  switch (status) {
    case DocumentStatus.EDIT_CONFLICT:
      return (
        <Note noteType="warning" title={`There is a new version of this ${entityLabel}`}>
          A new version of this {entityLabel} was created. To view or edit, refresh your browser or
          open the {entityLabel} in a new tab. Your current changes may be lost.
          <div>
            <TextLink href={entityHref} target="_blank" rel="noopener noreferrer">
              View {entityLabel} in new tab
            </TextLink>
          </div>
        </Note>
      );
    case DocumentStatus.INTERNAL_SERVER_ERROR:
      return (
        <Note noteType="negative" title="Server error">
          Due to a server error, we could not process this {entityLabel}. The fields are temporarily
          locked so that you won’t lose any important changes.
        </Note>
      );
    case DocumentStatus.CONNECTION_ERROR:
      return (
        <Note noteType="negative" title="Connection error">
          It appears that you aren’t connected to internet at the moment. The fields are temporarily
          locked so that you won’t lose any important changes.
        </Note>
      );
    case DocumentStatus.ARCHIVED:
      return (
        <Note noteType="warning" title="Archived">
          This {entityLabel} is archived and cannot be modified. Please unarchive it to make any
          changes.
        </Note>
      );
    case DocumentStatus.DELETED:
      return (
        <Note noteType="negative" title="Deleted">
          This {entityLabel} has been deleted and cannot be modified anymore
        </Note>
      );
    case DocumentStatus.NOT_ALLOWED:
      return (
        <Note noteType="warning" title="Read-only">
          You have read-only access to this {entityLabel}. If you need to edit it please contact
          your administrator.
        </Note>
      );
    case DocumentStatus.DEFAULT_LOCALE_FILE_ERROR:
      return (
        <Note noteType="negative" title="Asset missing">
          This asset is missing a file for the default locale
        </Note>
      );
    case DocumentStatus.LOCALE_VALIDATION_ERRORS:
      return (
        <Note noteType="negative" title="Validation error">
          The following locales have fields with errors: {formatErroredLocales(erroredLocales)}
        </Note>
      );
    default:
      return null;
  }
};

StatusCodeNotification.propTypes = {
  status: PropTypes.oneOf(Object.values(DocumentStatus)).isRequired,
  entityLabel: PropTypes.oneOf(['entry', 'asset']).isRequired,
  entityHref: PropTypes.string,
  erroredLocales: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      default: PropTypes.bool.isRequired,
      internal_code: PropTypes.string.isRequired,
    }).isRequired
  ),
};

export default StatusCodeNotification;
