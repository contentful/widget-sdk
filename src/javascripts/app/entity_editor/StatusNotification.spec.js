import React from 'react';
import Enzyme from 'enzyme';
import StatusNotification from './StatusNotification';
import DocumentStatusCode from 'data/document/statusCode';
import { range } from 'lodash';

describe('StatusNotification', () => {
  const statusCodes = [
    DocumentStatusCode.INTERNAL_SERVER_ERROR,
    DocumentStatusCode.CONNECTION_ERROR,
    DocumentStatusCode.ARCHIVED,
    DocumentStatusCode.DELETED,
    DocumentStatusCode.NOT_ALLOWED,
    DocumentStatusCode.DEFAULT_LOCALE_FILE_ERROR
  ];
  const erroredLocales = [
    { name: 'English (United States)', internal_code: 'en-US', default: false },
    { name: 'Spanish (Spain)', internal_code: 'es-ES', default: true },
    { name: 'French (France)', internal_code: 'fr-FR', default: false },
    { name: 'Arabic (Egypt)', internal_code: 'ar-EG', default: false },
    { name: 'German (Germany)', internal_code: 'de-DE', default: false }
  ];
  const props = {
    entityLabel: 'entry'
  };
  const render = () => Enzyme.shallow(<StatusNotification {...props} />);

  for (const status of statusCodes) {
    describe(status, () => {
      beforeEach(() => {
        props.status = status;
      });

      for (const entityLabel of ['entry', 'asset']) {
        describe(`when the entity label is "${entityLabel}"`, () => {
          beforeEach(() => {
            props.entityLabel = 'entry';
          });

          it('matches snapshot', () => {
            expect(render()).toMatchSnapshot();
          });
        });
      }
    });
  }

  describe('DocumentStatusCode.LOCALE_VALIDATION_ERRORS', () => {
    beforeEach(() => {
      props.status = DocumentStatusCode.LOCALE_VALIDATION_ERRORS;
    });

    for (const n of range(1, 6)) {
      describe(`when the number of errored locales is ${n}`, () => {
        beforeEach(() => {
          props.erroredLocales = erroredLocales.slice(0, n);
        });

        it('matches snapshot', () => {
          expect(render()).toMatchSnapshot();
        });
      });
    }
  });
});
