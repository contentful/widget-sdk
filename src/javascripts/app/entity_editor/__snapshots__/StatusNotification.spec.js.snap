// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusNotification DocumentStatusCode.LOCALE_VALIDATION_ERRORS when the number of errored locales is 1 matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Validation error
        </div>
        <div>
          The following locales have fields with errors: 
          English (United States)
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification DocumentStatusCode.LOCALE_VALIDATION_ERRORS when the number of errored locales is 2 matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Validation error
        </div>
        <div>
          The following locales have fields with errors: 
          Spanish (Spain) and English (United States)
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification DocumentStatusCode.LOCALE_VALIDATION_ERRORS when the number of errored locales is 3 matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Validation error
        </div>
        <div>
          The following locales have fields with errors: 
          Spanish (Spain), English (United States), and French (France)
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification DocumentStatusCode.LOCALE_VALIDATION_ERRORS when the number of errored locales is 4 matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Validation error
        </div>
        <div>
          The following locales have fields with errors: 
          Spanish (Spain), Arabic (Egypt), English (United States), and 1 other
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification DocumentStatusCode.LOCALE_VALIDATION_ERRORS when the number of errored locales is 5 matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Validation error
        </div>
        <div>
          The following locales have fields with errors: 
          Spanish (Spain), Arabic (Egypt), German (Germany), and 2 others
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification archived when the entity label is "asset" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--warning___3X53I"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--warning___39Bnz"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Archived
        </div>
        <div>
          This 
          asset
           is archived and cannot be modified. Please unarchive it to make any changes.
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification archived when the entity label is "entry" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--warning___3X53I"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--warning___39Bnz"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Archived
        </div>
        <div>
          This 
          entry
           is archived and cannot be modified. Please unarchive it to make any changes.
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification default-locale-file-error when the entity label is "asset" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Asset missing
        </div>
        <div>
          This asset is missing a file for the default locale
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification default-locale-file-error when the entity label is "entry" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Asset missing
        </div>
        <div>
          This asset is missing a file for the default locale
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification deleted when the entity label is "asset" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Deleted
        </div>
        <div>
          This 
          asset
           has been deleted and cannot be modified anymore
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification deleted when the entity label is "entry" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Deleted
        </div>
        <div>
          This 
          entry
           has been deleted and cannot be modified anymore
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification edit-conflict when the entity label is "asset" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--warning___3X53I"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--warning___39Bnz"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          There is a new version of this asset
        </div>
        <div>
          A new version of this 
          asset
           was created. To view or edit, refresh your browser or open the 
          asset
           in a new tab. Your current changes may be lost.
          <div>
            <a
              class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
              data-test-id="cf-ui-text-link"
              href="hrefToTheEntityDetailsPage"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="TabFocusTrap__TabFocusTrap___39Vty"
                tabindex="-1"
              >
                View 
                asset
                 in new tab
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification edit-conflict when the entity label is "entry" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--warning___3X53I"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--warning___39Bnz"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          There is a new version of this entry
        </div>
        <div>
          A new version of this 
          entry
           was created. To view or edit, refresh your browser or open the 
          entry
           in a new tab. Your current changes may be lost.
          <div>
            <a
              class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
              data-test-id="cf-ui-text-link"
              href="hrefToTheEntityDetailsPage"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span
                class="TabFocusTrap__TabFocusTrap___39Vty"
                tabindex="-1"
              >
                View 
                entry
                 in new tab
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification editing-not-allowed when the entity label is "asset" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--warning___3X53I"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--warning___39Bnz"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Read-only
        </div>
        <div>
          You have read-only access to this 
          asset
          . If you need to edit it please contact your administrator.
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification editing-not-allowed when the entity label is "entry" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--warning___3X53I"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--warning___39Bnz"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Read-only
        </div>
        <div>
          You have read-only access to this 
          entry
          . If you need to edit it please contact your administrator.
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification internal-server-error when the entity label is "asset" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Server error
        </div>
        <div>
          Due to a server error, we could not process this 
          asset
          . The fields are temporarily locked so that you won’t lose any important changes.
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification internal-server-error when the entity label is "entry" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Server error
        </div>
        <div>
          Due to a server error, we could not process this 
          entry
          . The fields are temporarily locked so that you won’t lose any important changes.
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification ot-connection-error when the entity label is "asset" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Connection error
        </div>
        <div>
          It appears that you aren’t connected to internet at the moment. The fields are temporarily locked so that you won’t lose any important changes.
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StatusNotification ot-connection-error when the entity label is "entry" matches snapshot 1`] = `
<body>
  <div>
    <div
      class="Note__Note___2eSKN Note__Note--negative___3fkwL"
      data-test-id="cf-ui-note"
    >
      <div
        class="Note__Note__icon___20RqC"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
          />
        </svg>
      </div>
      <div
        class="Note__Note__info___2m_-K"
      >
        <div
          class="Note__Note__title___2nwpx"
        >
          Connection error
        </div>
        <div>
          It appears that you aren’t connected to internet at the moment. The fields are temporarily locked so that you won’t lose any important changes.
        </div>
      </div>
    </div>
  </div>
</body>
`;
